app/models/library_entry.rb

```diff
@@ -187,16 +187,18 @@ def kind
     if status_changed? && completed?
       # update progress to the cap
       self.progress = media.progress_limit if media&.progress_limit
-      # set finished_at and started_at for the first consume
-      self.started_at ||= Time.now unless imported
-      self.finished_at ||= Time.now unless imported
     end
 
     unless imported
-      # When progress is changed, update progressed_at
-      self.progressed_at = Time.now if progress_changed? || status_changed?
-      # When marked current and started_at doesn't exist
-      self.started_at ||= Time.now if current?
+      self.progressed_at = Time.now if status_changed? || progress_changed?
+
+      if status_changed?
+        self.started_at ||= Time.now if current?
+        self.started_at ||= Time.now if completed?
+        self.finished_at ||= Time.now if completed?
+      end
+
+      self.status = :current if progress_changed? && !status_changed?
```

Comment:
Isn't this basically "set status to current if the progress has changed and the status hasn't" and won't that just constantly set things to current?

Reply:
is that not okay? or do we want to check that it's not current?

Reply:
Y'know, now that I think more it does seem like it would be fine 🤔